src/server/transaction.cc

```diff
@@ -1104,10 +1104,13 @@ bool Transaction::ScheduleInShard(EngineShard* shard, bool execute_optimistic) {
 
     // We need to acquire the fp locks because the executing callback
     // within RunCallback below might preempt.
-    bool keys_unlocked = GetDbSlice(shard->shard_id()).Acquire(mode, lock_args);
+    auto [keys_unlocked, has_duplicates] = GetDbSlice(shard->shard_id()).Acquire(mode, lock_args);
```

Comment:
when running dragonfly with lock on hash tag will this always be true for multi keys?
if so do you see any downside?

Reply:
I think we are missing some tests that check the response for mget for the same key
lets add this test before merging this pr

Reply:
good catch, i did not think of it. yes, there is a downside

Reply:
I think I am gonna drop the transaction optimization for now, too bad it does not work.